### ISSUE TITLE
extract_clim() Error in if (start_time < start) { : argument is of length zero

### ISSUE THREAD

**Issue by manmatej** [dklinges9/mcera5#38]
Hi, I am trying to extract climate data from ERA5 .nc files I downloaded in March 2024.
I am getting error with time.

 "Error in if (start_time < start) { : argument is of length zero"

I am a bit lost in cds versions and compatibility. 

I tried both with `main` and `developer` versions of pacakges but still the same Error. Am I doing something wrong? 
[Here is my nc file. ](https://1drv.ms/u/s!AuTei9SXVxs_hvZ59jvbjhExSoDc4A?e=X08IAb)
 
Thank you for any help. 

```
remotes::install_github("dklinges9/mcera5") # main
remotes::install_github("dklinges9/mcera5", ref = "new_cds_time_edits") #dev 

library(mcera5)
library(terra)
library(ncdf4)

# explore era5 file with terra pacakge
# looks OK
r<-rast("2023-1712168514.5906076-24507-16-90f6131f-f79e-4840-91b0-b0ab86449efd.nc")
varnames(r)
longnames(r)
names(r)
time(r)
r1<-r[[1]]
plot(r1)

# prepare data frame for micropoint librarry
my_nc <- "2023-1712168514.5906076-24507-16-90f6131f-f79e-4840-91b0-b0ab86449efd.nc"

# i am sure this point is within my extent
x <- 13.8084500 
y <- 50.4111411

# i am sure this time is covered by my file
st_time <- as.POSIXct("2023-01-01 01:00", tz = "UTC")
en_time <- as.POSIXct("2023-12-31 23:00", tz = "UTC")

clim_point <- extract_clim(nc = my_nc, long = x, lat = y,
                           start_time = st_time, end_time = en_time,
                           format = "micropoint",
                           cds_version = "legacy")
```

Error in if (start_time < start) { : argument is of length zero

------------------------------------------------------
Here my sessioninfo

```
R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=Czech_Czechia.utf8  LC_CTYPE=Czech_Czechia.utf8    LC_MONETARY=Czech_Czechia.utf8
[4] LC_NUMERIC=C                   LC_TIME=Czech_Czechia.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ncdf4_1.22       terra_1.7-71     micropoint_0.1.0 mcera5_0.4.0    

loaded via a namespace (and not attached):
[1] compiler_4.2.0    magrittr_2.0.3    tools_4.2.0       rstudioapi_0.16.0 Rcpp_1.0.13      
[6] remotes_2.5.0     codetools_0.2-18 
```

**Comment by dklinges9** [comment]
Hi Matěj,

Just letting you know that indeed this was a genuine bug, that I'm working on fixing now. This also prompted me to simplify things for users for the future as well. Update will come soon!

**Comment by dklinges9** [comment]
Hi Matěj,

I gave the recent changes a bit more thought and decided that it would be easier for users to not have them worry about whether their ERA5 data was from the old or new CDS. So I now handle such matters completely internally. With the [most recent commit](04bf9e57fb86eff4c951acd3c3fc87769f8ed25a) there is no longer an argument `cds_version`, and the package figures out how to handle the time dimension from the netCDF itself.

I was able to successfully `extract_clim()` from your attached netCDF with your code. Install the newest version of the `master` branch and give it a try now,  and let me know if you face any issues!

Best, Dave
